Extensions/dnSpy.Debugger/dnSpy.Debugger.DotNet.CorDebug/Impl/ModuleCreator.cs

```diff
@@ -92,10 +92,18 @@ static class ModuleCreator {
 		static Func<DmdLazyMetadataBytes> CreateDynamicGetMetadataDelegate(DbgEngineImpl engine, DnModule dnModule) {
 			Debug.Assert(dnModule.IsDynamic);
 			var comMetadata = dnModule.CorModule.GetMetaDataInterface<IMetaDataImport2>();
-			if (comMetadata is null)
-				throw new InvalidOperationException();
-			var result = new DmdLazyMetadataBytesCom(comMetadata, engine.GetDynamicModuleHelper(dnModule), engine.DmdDispatcher);
-			return () => result;
+			if (comMetadata == null) {
+				if (dnModule.Process.CorProcess.CLRVersion.Major != 6) {
+					throw new InvalidOperationException();
+				}
+				else  //Net 6 hack for Issue: https://github.com/dnSpyEx/dnSpy/issues/96
+					return () => new DmdLazyMetadataBytesNull();
```

Comment:
```suggestion
				if (dnModule.Process.CorProcess.CLRVersion.Major == 6) {
					// Workaround for a .NET 6 bug where dynamic modules don't have a IMetaDataImport2 interface when a debugger is attached
					// see https://github.com/dnSpyEx/dnSpy/issues/96
					return () => new DmdLazyMetadataBytesNull();
				}

				throw new InvalidOperationException();
```
Reason: Inverted `if` for clarity, `else` clause removed as it was redundant. Comment updated with additional details.